test/integration/highlight_editor_spec.mjs

```diff
@@ -2144,4 +2145,276 @@ describe("Highlight Editor", () => {
       );
     });
   });
+
+  describe("Annotation pop-up has the expected behaviour", () => {
```

Comment:
Again, a strange test name. (And please be consistent, use *either* `popup` or `pop-up` consistently throughout the patch.)